src/network/networkd-network.c

```diff
@@ -36,7 +36,12 @@ static int network_config_compare_func(const void *a, const void *b) {
         if (r != 0)
                 return r;
 
-        return y->line - x->line;
+        if (x->line < y->line)
+                return -1;
+        if (x->line > y->line)
+                return 1;
```

Comment:
Hmm, that's an interesting bug. So the comparison function was borked all the time and nobody noticed.